api.planx.uk/modules/analytics/metabase/shared/client.ts

```diff
@@ -0,0 +1,155 @@
+import axios from "axios";
+import type {
+  AxiosInstance,
+  AxiosError,
+  AxiosRequestConfig,
+  InternalAxiosRequestConfig,
+} from "axios";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+// Custom error class for Metabase-specific errors
+export class MetabaseError extends Error {
+  constructor(
+    message: string,
+    public statusCode?: number,
+    public response?: unknown,
+  ) {
+    super(message);
+    this.name = "MetabaseError";
+  }
+}
+
+interface MetabaseConfig {
+  baseURL: string;
+  apiKey: string;
+  timeout?: number;
+  retries?: number;
+}
+
+// Validate environment variables
+const validateConfig = (): MetabaseConfig => {
+  const baseURL = process.env.METABASE_BASE_URL;
+  const apiKey = process.env.METABASE_API_KEY;
+
+  if (!baseURL) {
+    throw new Error(
+      "METABASE_BASE_URL is not defined in environment variables",
+    );
+  }
+  if (!apiKey) {
+    throw new Error("METABASE_API_KEY is not defined in environment variables");
+  }
+
+  return {
+    baseURL,
+    apiKey,
+    timeout: parseInt(process.env.METABASE_TIMEOUT ?? "30000"), // 30 second default timeout
+    retries: parseInt(process.env.METABASE_MAX_RETRIES ?? "3"), // 3 default retries
+  };
+};
+
+// Extended request config to include retry count
+interface ExtendedAxiosRequestConfig extends InternalAxiosRequestConfig {
+  retryCount?: number;
+}
+
+// Create and configure Axios instance
+export const createMetabaseClient = (): AxiosInstance => {
+  const config = validateConfig();
+
+  const client = axios.create({
+    baseURL: config.baseURL,
+    headers: {
+      "X-API-Key": config.apiKey,
+      "Content-Type": "application/json",
+    },
+    timeout: config.timeout,
+  });
+
+  // Request interceptor
+  client.interceptors.request.use(
+    (config) => {
+      // Could add request logging here
+      return config;
+    },
+    (error) => {
+      return Promise.reject(error);
+    },
+  );
```

Comment:
Is this actually doing anything currently?

Reply:
No, I think I used it before for debugging but will delete now as it's basically the same variety of redundant code as leftover `console.log`s